src/content/en/updates/2017/10/animated-blur.md

```diff
@@ -0,0 +1,189 @@
+
+## The problem
+
+<img src="/web/updates/images/2017/10/animated-blur/pipeline.png" alt="Markup is
+  turned into textures by the CPU. Textures are uploaded to the GPU. The GPU
+  draws these textures to the framebuffer using shaders. The blurring happens in
+  the shader." class="attempt-right">
+
+As of now, we can't make animating a blur work efficiently. We can, however,
+find a work-around that looks *good enough*, but is, technically speaking, not
+an animated blur. To get started, let's first understand *why* the animated blur
+is slow. To blur elements on the web there're two techniques: The CSS `filter`
+property and SVG filters. Thanks to increased support and ease of use, CSS
+filters are typically used. However, if you are required to support Internet
+Explorer, you have no choice but to use SVG filters as IE 10 and 11 support
+those but not CSS filters. The good news is that our workaround for animating a
+blur works with both techniques. So let's try to find the bottleneck by looking
+at DevTools:
+
+If you enable "Paint Flashing" in DevTools, it does not flash at all. It looks
+like no repaints are happening. And it's technically correct as a "repaint"
+refers to the CPU having to repaint the texture of an promoted element. Whenever
+an element is both promoted *and* blurred, the blur is applied by the GPU using
+a shader.
+
+Both SVG filters and CSS filters use [convolution
+filters](https://en.wikipedia.org/wiki/Kernel_(image_processing)) to apply a
+blur. Convolution filters are fairly expensive as for every output pixel a
+number of input pixels have to be considered. The bigger the image or the bigger
+the blur radius, the more costly the effect is.
+
+And that's where the problem lies, we are running a rather expensive GPU
+operation every frame, blowing our frame budget of 16ms and therefore ending up
+well below 60fps.
+
+
+## Down the rabbit hole
+
+So what can we do to make this run smoothly? We can use sleight of hand! Instead
+of animating the actual blur value (the radius of the blur), we pre-compute a
+couple of blurred copies where the blur value increases exponentially and
+cross-fade between them using `opacity`.
+
+The cross-fade is a series of overlapping opacity fade-ins and fade-outs. If we
+have four blur stages for example, we fade out the first stage while fading in
+the second stage at the same time. Once the second stage reaches 100% opacity
+and the first one has reached 0%, we fade out the second stage while fading in
+the third stage. Once that is done, we finally fade out the third stage and fade
+in the fourth and final version. In this scenario, each stage would take ¼ of
+the total desired duration. Visually, this looks very similar to a real,
+animated blur.
+
+In our experiments increasing the blur radius exponentially per stage yielded
+the best visual results. Example: If we have four blur stages we'd apply
+`filter: blur(2^n)` to each stage, i.e. stage 0: 1px, stage 1: 2px, stage 2: 4px
```

Comment:
should some of `stage 0: 1px` be code?

Reply:
No, I don’t think so. It’s just the parameter for the `blur()`.